internal/volumes/volumes.go

```diff
@@ -391,11 +396,27 @@ func GetCacheMount(args []string, _ storage.Store, _ string, additionalMountPoin
 		}
 
 		if id != "" {
-			newMount.Source = filepath.Join(cacheParent, filepath.Clean(id))
-			buildahLockFilesDir = filepath.Join(BuildahCacheLockfileDir, filepath.Clean(id))
+			newSrc, err := securejoin.SecureJoin(cacheParent, filepath.Clean(id))
```

Comment:
We don't need to be using the ID to construct this path, at all.  Try something else, like running it through `digest.Canonical.FromString()` slicing off the first N characters in the result's `Encoded()` value, and using that.